Added Prio and PrioID

pscn opened the pull request:
See: https://www.musicpd.org/doc/html/protocol.html

Unfortunately I'm not sure since which protocol version this is supported :(

fhs commented on mpd/client.go:
```diff
@@ -360,6 +360,31 @@ func (c *Client) PlaylistInfo(start, end int) ([]Attrs, error) {
+// Prio set the priority of the specified songs. If end is negative but start
+// is positive, it does it for the song at position start. If both start and
+// end are positive, it does it for positions in range [start, end).
+func (c *Client) Prio(prio, start, end int) error {
```
Maybe name this function SetPriority? There is no reason we have to follow MPD's obscure naming.

fhs commented on mpd/client.go:
```diff
@@ -360,6 +360,31 @@ func (c *Client) PlaylistInfo(start, end int) ([]Attrs, error) {
+}
+
+// PrioID set the prio of the song with the given id
+func (c *Client) PrioID(prio, id int) error {
```
Name this SetPriorityID?

fhs commented on mpd/client.go:
```diff
@@ -360,6 +360,31 @@ func (c *Client) PlaylistInfo(start, end int) ([]Attrs, error) {
 }
 
+// Prio set the priority of the specified songs. If end is negative but start
+// is positive, it does it for the song at position start. If both start and
```
Non-actionable comment: >= 0 mean non-negative, not "positive". I realize the same wording is used in other functions, so don't worry about it.

pscn commented:
Renaming done.  I missed the part where you do it for volume like this.  I hope I come around to the unit tests later today.

pscn commented:
@fhs should I add SetPriority and SetPriorityID to the CommandList?

fhs commented:
> @fhs should I add SetPriority and SetPriorityID to the CommandList?

Sure, if MPD supports it and it works.

pscn commented:
Added it to CommandList also, as my MPD supports it.  Also fixed an issue with the delete call in CommandList, please let me know if I should do this in a separate pull request.

fhs commented on mpd/internal/server/server.go:
```diff
@@ -387,6 +387,66 @@ func (s *server) writeResponse(p *textproto.Conn, args []string, okLine string)
+			// Updating a range of the playlist from specified start/end positions.
+			start, err = strconv.Atoi(rng[0])
+			if err != nil {
+				ack("Integer or range expected")
```
don't captitalize error messages (ditto for errors below)

pscn replied:
That was copied from another case where a range is extracted.  Should I fix those to while I'm at it?

fhs commented on mpd/internal/server/server.go:
```diff
@@ -387,6 +387,66 @@ func (s *server) writeResponse(p *textproto.Conn, args []string, okLine string)
+				ack("invalid song position")
+				return
+			}
+			start = i
```
check `start` is less than `s.currentPlaylist.Len()` and >= 0?

pscn replied:
Checking for start >=0 is sensible, but I don't want to test for s.currentPlaylist.Len() as it doesn't matter for this case.

pscn commented:
I'm failing in using git and my branch is all kinds of messed up :'( Don't really have the time to figure out the correct way to resolve this, so I'm closing this for now and start a clean pull request again later.  Sorry for the waste of time :|
